pkg/asset/installconfig/baremetal/validation.go

```diff
@@ -20,3 +20,19 @@ func ValidateProvisioning(ic *types.InstallConfig) error {
 
 	return allErrs.ToAggregate()
 }
+
+func ValidateStaticBootstrapNetworking(ic *types.InstallConfig) error {
+	if ic.Platform.BareMetal == nil {
```

Comment:
With this change, a check for ic.Platform.BareMetal == nil will take place twice from  https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/platformprovisioncheck.go#L75.
We could separate the platform check into its own validation and call it from  https://github.com/openshift/installer/blob/master/pkg/asset/installconfig/platformprovisioncheck.go#L75?